python/pyarrow/_fs.pyx

```diff
@@ -335,6 +467,21 @@ cdef class FileSystem(_Weakrefable):
         tuple of (FileSystem, str path)
             With (filesystem, path) tuple where path is the abstract path
             inside the FileSystem instance.
+
+        Examples
+        --------
+        Create a new FileSystem subclass from path:
+
+        >>> local_new, path = fs.LocalFileSystem().from_uri('file:///tmp/fileinfo.dat')
```

Comment:
```suggestion
        >>> local_new, path = fs.LocalFileSystem.from_uri('file:///tmp/fileinfo.dat')
```

Since this is a staticmethod, you actually don't need to call it from an instance (as you already did for the S3 example below).

Reply:
And actually it's maybe also better (or at least shorter) to use `FileSystem.from_uri` instead of `LocalFileSystem.from_uri`. 

There is nothing about this method that is specific to a FileSystem subclass, and so it is not because you call it from `LocalFileSystem` that it is guaranteed to return a local filesystem (eg if you would pass a S3 URI to the LocalFileSystem, it would (somewhat confusingly) also work fine)